src/components/NodeTotalChart/CustomTick.tsx

```diff
@@ -10,20 +13,63 @@ const StyledLabel = styled("tspan")({
   userSelect: "none",
 });
 
-const CustomTick = ({ x, y, payload }) => {
-  const textLines = payload.value?.replace(/_/g, " ").split(" ");
+/**
+ * Renders a custom X-Axis tick for the NodeTotalChart
+ * which handles the rotation and formatting of the tick label
+ *
+ * @note Details on how the `angled` prop is implemented:
+ * - If `true`, the tick label will be rotated 65 degrees
+ *   and will be displayed in a single line. Any excess
+ *   characters (`labelLength`) will be trimmed and replaced with an ellipsis.
+ * - If `false`, the tick label will be displayed in multiple lines
+ *   denoted by spaces in the label string.
+ * @returns {React.FC}
+ */
+const CustomTick = ({
+  x,
+  y,
+  payload,
+  labelLength = 7,
+  angled = false,
+  handleMouseEnter = null,
+  handleMouseLeave = null,
+}) => {
+  const label: string = titleCase(payload?.value?.replace(/_/g, " ")?.trim() || "");
+  const labelLines: string[] = useMemo<string[]>(() => {
+    if (angled) {
+      return [label.length > labelLength ? `${label.slice(0, labelLength).trim()}...` : label];
+    }
+
+    return label.split(" ");
+  }, [label, angled, labelLength]);
+
+  const onMouseEnter = useCallback(() => {
+    handleMouseEnter?.({ x, y, label: titleCase(label) });
+  }, [x, y, label, handleMouseEnter]);
+
+  const onMouseLeave = useCallback(() => {
+    handleMouseLeave?.();
+  }, [x, y, label, handleMouseLeave]);
 
   return (
-    <g transform={`translate(${x},${y})`}>
-      <text x={0} y={0} dy={6}>
-        {textLines.map((tspan, index) => (
-          <StyledLabel key={tspan} textAnchor="middle" x={0} dy={6 * index + 8}>
-            {tspan}
-          </StyledLabel>
-        ))}
-      </text>
+    <g transform={`translate(${x},${y})`} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
+      <g transform={angled ? "rotate(65), translate(-4, 7)" : undefined}>
```

Comment:
Is there a purpose for the translate? If so, we can probably disregard. But in extreme examples, in this case 50 bar charts, this does cause an misalignment. See screenshots:

<details><summary>With translate</summary>

![image](https://github.com/user-attachments/assets/85b1e35f-49d5-4e10-8ee3-7187f4698a13)
</details>

<details><summary>Without translate</summary>

![Screenshot 2024-08-15 at 12 56 12 PM](https://github.com/user-attachments/assets/e971face-d581-489b-8cff-1102c8404316)
</details>

Reply:
Honestly...I'm not sure what I used it for originally or why I never checked without it. Thanks for finding this! a912b92